test/e2e/postinstall/syncsets/syncsets_suite_test.go

```diff
@@ -0,0 +1,517 @@
+
+				By("Deleting the syncset with upsert apply mode")
+				err = hiveClient.Delete(ctx, syncSetWithUpsertApplyMode)
+				Ω(err).ShouldNot(HaveOccurred())
+
+				time.Sleep(5 * time.Second)
+
+				By("Verifying the ConfigMap was not deleted on the target cluster")
+				resultConfigMap = &corev1.ConfigMap{}
+				err = targetClusterClient.Get(ctx, client.ObjectKey{Name: "foo", Namespace: "default"}, resultConfigMap)
+				Ω(err).ShouldNot(HaveOccurred())
+				Ω(resultConfigMap.Data).Should(Equal(map[string]string{"foo": "bar"}))
+
+			})
+
+			It("Test SyncSet patches", func() {
+				ctx := context.TODO()
+				configMap := &corev1.ConfigMap{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "ConfigMap",
+						APIVersion: "v1",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "foo",
+						Namespace: "default",
+					},
+					Data: map[string]string{
+						"foo": "bar",
+					},
+				}
+				SyncSetPatch := &hivev1.SyncSet{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-syncpatch",
+						Namespace: clusterNamespace,
+					},
+					Spec: hivev1.SyncSetSpec{
+						ClusterDeploymentRefs: []corev1.LocalObjectReference{
+							{
+								Name: clusterName,
+							},
+						},
+						SyncSetCommonSpec: hivev1.SyncSetCommonSpec{
+							ResourceApplyMode: "Sync",
+							Patches: []hivev1.SyncObjectPatch{
+								{
+									APIVersion: "v1",
+									Kind:       "ConfigMap",
+									Name:       "foo",
+									Namespace:  "default",
+									Patch:      `{ "data": { "foo": "new-bar" } }`,
+									PatchType:  "merge",
+								},
+							},
+						},
+					},
+				}
+				By("Creating a resource ConfigMap on target cluster")
+				err := targetClusterClient.Create(ctx, configMap)
+				Ω(err).ShouldNot(HaveOccurred())
+				time.Sleep(5 * time.Second)
+
+				By("Verify resource ConfigMap created on target cluster")
+				resultConfigMap := &corev1.ConfigMap{}
+				err = targetClusterClient.Get(ctx, client.ObjectKey{Name: "foo", Namespace: "default"}, resultConfigMap)
+				Ω(err).ShouldNot(HaveOccurred())
+				Ω(resultConfigMap.Data).Should(Equal(map[string]string{"foo": "bar"}))
+
+				By("Creating a patch with merge patchType")
+				err = hiveClient.Create(ctx, SyncSetPatch)
+				Ω(err).ShouldNot(HaveOccurred())
+
+				err = waitForSyncSetInstanceApplied("test-syncpatch", "syncset", clusterNamespace)
+				Ω(err).ShouldNot(HaveOccurred())
+
+				By("Verifying the resource was patched")
+				resultConfigMap = &corev1.ConfigMap{}
+				err = targetClusterClient.Get(ctx, client.ObjectKey{Name: "foo", Namespace: "default"}, resultConfigMap)
+				Ω(err).ShouldNot(HaveOccurred())
+				Ω(resultConfigMap.Data).Should(Equal(map[string]string{"foo": "new-bar"}))
+
+				By("Modifying the patch type to strategic")
+				patch := `{ "data": { "foo": "baz-strategic" } }`
+				patchType := "strategic"
+				modifySynSetPatchType(hiveClient, SyncSetPatch, patch, patchType)
```

Comment:
It may be better to delete the previous syncset and create a new one. That way when .Status.Applied becomes true, then you are sure that it has been truly applied. See comment on line 217.
To create the new syncset, you could change line 162 to be a function that takes a patch and patchType instead (as in line 59)